api/controllers/content/movie-detail.js

```diff
@@ -15,14 +15,23 @@ module.exports = {
   exits: {
     success: {
       viewTemplatePath: 'content/movieDetail'
+    },
+    redirect: {
+      responseType: 'redirect'
     }
   },
 
 
   fn: async function (inputs) {
-    let req = this.req;
-    let filmId = req.params.id;
-    let film = await Film.findOne({ id: filmId }).meta({ skipRecordVerification: true });
+    // find the user that is logged in 
+    let sessionUserId = this.req.session.userId;
+    let user = await User.findOne({id: sessionUserId})
+
+    // find the film                                        // this so that the doesn't throw warnings
+    let film = await Film.findOne({ id: this.req.params.id }).meta({ skipRecordVerification: true });
+
+    // check if the user and the movie have age restrictions
+    if (user.isKid != undefined && film.properForKids == false) throw {redirect: '/kidsmovies'};
```

Comment:
removed some not needed lines and added comments so that is can be easier to understand.